From: Daniel Hartwig <mandyke@gmail.com>
Date: Wed, 23 Jul 2014 17:42:14 -0400
Subject: r673_zeroconf-nodups

---
 source/src/zeroconf.c |   24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/source/src/zeroconf.c b/source/src/zeroconf.c
index df41d96..a251426 100644
--- a/source/src/zeroconf.c
+++ b/source/src/zeroconf.c
@@ -129,6 +129,8 @@ static time_t fd_last_used(int fd, time_t clip_time) {
     return now - ft;
 }
 
+static void remove_duplicate_services(struct daemon_data *d);
+
 /* Write host data to host file */
 static int write_hosts(struct daemon_data *d) {
     struct host *h;
@@ -152,6 +154,8 @@ static int write_hosts(struct daemon_data *d) {
         return -1;
     }
 
+    remove_duplicate_services(d);
+
     for (h = d->hosts; h; h = h->next) {
         char t[256], a[AVAHI_ADDRESS_STR_MAX];
 
@@ -190,6 +194,26 @@ static void free_host(struct host *h) {
     free(h);
 }
 
+/* Remove hosts with duplicate service names from the host list.
+ * Hosts with multiple IP addresses show up more than once, but
+ * should all have the same service name: "distcc@hostname" */
+static void remove_duplicate_services(struct daemon_data *d) {
+    struct host *host1, *host2, *prev;
+    assert(d);
+    for (host1 = d->hosts; host1; host1 = host1->next) {
+        assert(host1->service);
+        for (host2 = host1->next, prev = host1; host2; host2 = prev->next) {
+            assert(host2->service);
+            if (!strcmp(host1->service, host2->service)) {
+                prev->next = host2->next;
+                free_host(host2);
+            } else {
+                prev = host2;
+            }
+        }
+    }
+}
+
 /* Remove a service from the host list */
 static void remove_service(struct daemon_data *d, AvahiIfIndex interface, AvahiProtocol protocol, const char *name, const char *domain) {
     struct host *h, *p = NULL;
